storage: TestLogGrowthWhenRefreshingPendingCommands failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/04cba2800919bdcf6a8467e8da97ae44b77a9626

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stressrace TESTS=TestLogGrowthWhenRefreshingPendingCommands PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=975334&tab=buildLog

```
I181019 06:35:17.462601 57331 storage/replica_command.go:816  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n4,s4):4): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
I181019 06:35:17.464662 57331 storage/replica.go:3884  [s1,r1/1:/M{in-ax},txn=396597c7] proposing ADD_REPLICA((n4,s4):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4] next=5
I181019 06:35:17.465077 57359 storage/replica.go:3884  [s1,r1/1:/M{in-ax},txn=396597c7] proposing ADD_REPLICA((n4,s4):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4] next=5
I181019 06:35:17.466576 57331 storage/store_snapshot.go:621  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 3cd82969 at applied index 26
I181019 06:35:17.466978 57331 storage/store_snapshot.go:664  [s1,r1/1:/M{in-ax}] streamed snapshot to (n5,s5):?: kv pairs: 60, log entries: 16, rate-limit: 2.0 MiB/sec, 1ms
I181019 06:35:17.467301 57807 storage/replica_raftstorage.go:804  [s5,r1/?:{-}] applying preemptive snapshot at index 26 (id=3cd82969, encoded size=12094, 1 rocksdb batches, 16 log entries)
I181019 06:35:17.468196 57807 storage/replica_raftstorage.go:810  [s5,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I181019 06:35:17.468509 57331 storage/replica_command.go:816  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n5,s5):5): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):4, next=5, gen=0]
I181019 06:35:17.470278 57331 storage/replica.go:3884  [s1,r1/1:/M{in-ax},txn=52031f95] proposing ADD_REPLICA((n5,s5):5): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4 (n5,s5):5] next=6
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (1.02s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/proposeOnFollower=true
W181019 06:35:18.171311 58152 storage/store.go:1490  [s4,r1/4:/M{in-ax}] could not gossip node liveness: periodic gossip is disabled
W181019 06:35:18.171414 58151 storage/store.go:1490  [s4,r1/4:/M{in-ax}] could not gossip system config: periodic gossip is disabled
W181019 06:35:18.173481 58246 storage/store.go:1490  [s5,r1/5:/M{in-ax}] could not gossip system config: periodic gossip is disabled
W181019 06:35:18.173504 58247 storage/store.go:1490  [s5,r1/5:/M{in-ax}] could not gossip node liveness: periodic gossip is disabled
I181019 06:35:18.177036 58067 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] txnHeartbeat: aborting txn
I181019 06:35:18.177480 58159 storage/node_liveness.go:790  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context canceled)
W181019 06:35:18.177520 58159 storage/node_liveness.go:454  [hb] failed node liveness heartbeat: context canceled
W181019 06:35:18.177586 58254 kv/txn_interceptor_heartbeat.go:538  node unavailable; try another peer
I181019 06:35:18.177639 58059 internal/client/txn.go:637  async rollback failed: node unavailable; try another peer
W181019 06:35:18.177650 58254 storage/node_liveness.go:454  [hb] failed node liveness heartbeat: context canceled
I181019 06:35:18.427938 58166 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] storage.raftlog: processing replica
E181019 06:35:18.428058 57953 storage/queue.go:791  [raftlog,s1,r1/1:/M{in-ax}] result is ambiguous (server shutdown)
W181019 06:35:18.428883 57924 storage/raft_transport.go:584  while processing outgoing Raft queue to node 2: EOF:
W181019 06:35:18.428884 57909 storage/raft_transport.go:584  while processing outgoing Raft queue to node 5: EOF:
W181019 06:35:18.428929 58052 storage/raft_transport.go:584  while processing outgoing Raft queue to node 3: EOF:
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/proposeOnFollower=true (0.26s)
    	client_raft_test.go:1266: raft log size grew to 51 KiB
    	client_raft_test.go:1266: raft log size grew to 60 KiB
    	client_raft_test.go:1266: raft log size grew to 62 KiB
    	client_raft_test.go:1266: raft log size grew to 63 KiB
    	client_raft_test.go:1264: raft log size grew to 65 KiB

```

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -1255,12 +1255,17 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 					t.Fatalf("raft leader should be node 0, but got status %+v", status)
 				}
 
-				// Check raft log size.
+				// Check the raft log size. We allow GetRaftLogSize to grow up
+				// to twice RaftMaxUncommittedEntriesSize because its total
+				// includes a little more state (the roachpb.Value checksum,
+				// etc.). The important thing here is that the log doesn't grow
+				// forever.
+				logSizeLimit := int64(2 * sc.RaftMaxUncommittedEntriesSize)
 				curlogSize := leaderRepl.GetRaftLogSize()
 				logSize := curlogSize - initLogSize
 				logSizeStr := humanizeutil.IBytes(logSize)
 				// Note that logSize could be negative if something got truncated.
-				if logSize > int64(sc.RaftMaxUncommittedEntriesSize) {
+				if logSize > logSizeLimit {
 					t.Fatalf("raft log size grew to %s", logSizeStr)
 				}
 				t.Logf("raft log size grew to %s", logSizeStr)

